pkg/envoy/rds.go

```diff
@@ -94,21 +95,35 @@ func (l RouteDomainCheck) Description() string {
 	return fmt.Sprintf("Checking whether %s is configured with correct %s Envoy route", l.ConfigGetter.GetObjectName(), l.RouteName)
 }
 
-// NewOutboundRouteDomainPodCheck creates a DestinationEndpointCheck which checks whether the Envoy config has an outbound dynamic route domain to the Pod
-func NewOutboundRouteDomainPodCheck(configGetter ConfigGetter, pod *corev1.Pod) RouteDomainCheck {
-	return RouteDomainCheck{
-		ConfigGetter: configGetter,
-		RouteName:    OutboundDynamicRouteConfigName,
-		Domain:       fmt.Sprintf("%s.%s", pod.Name, pod.Namespace),
-	}
+// HasOutboundDynamicRouteConfigDomainPodCheck creates a new common.Runnable, which checks
+// whether the Envoy config has outbound dynamic route domains to the Pod's services.
+func HasOutboundDynamicRouteConfigDomainPodCheck(client kubernetes.Interface, configGetter ConfigGetter, pod *corev1.Pod) RouteDomainCheck {
+	return ArePodServicesInDynamicRouteConfigDomains(client, configGetter, pod, OutboundDynamicRouteConfigName)
+}
+
+// HasInboundDynamicRouteConfigDomainPodCheck creates a new common.Runnable, which checks
+// whether the Envoy config has inbound dynamic route domains from the Pod's services.
+func HasInboundDynamicRouteConfigDomainPodCheck(client kubernetes.Interface, configGetter ConfigGetter, pod *corev1.Pod) RouteDomainCheck {
+	return ArePodServicesInDynamicRouteConfigDomains(client, configGetter, pod, InboundDynamicRouteConfigName)
 }
 
-// NewInboundRouteDomainPodCheck creates a DestinationEndpointCheck which checks whether the Envoy config has an inbound dynamic route domain to the Pod
-func NewInboundRouteDomainPodCheck(configGetter ConfigGetter, pod *corev1.Pod) RouteDomainCheck {
+// ArePodServicesInDynamicRouteConfigDomains checks whether the pod's corresponding service's domains are
+// contained in the envoy dynamic route config domain list.
+func ArePodServicesInDynamicRouteConfigDomains(client kubernetes.Interface, configGetter ConfigGetter, pod *corev1.Pod, routeName string) RouteDomainCheck {
```

Comment:
could this name be simplified? `ArePodServicesInDynamicRouteConfigDomains`

Reply:
Hmmm I think we should leave this for now - I can't figure out a better name.

Reply:
How about `NewPodServicesRouteDomainCheck`  to parallel with `NewInboundRouteDomainPodCheck`

Reply:
or rather NewServicesRouteDomainPodCheck -- not sure which follows more logically

Reply:
Done! Chose `NewPodServicesRouteDomainCheck`